src/System.IO.Compression/src/System/IO/Compression/ZLibNative.cs

```diff
@@ -403,6 +403,38 @@ public ErrorCode InflateEnd()
                 return errC;
             }
 
+            /// <summary>
+            /// This function is equivalent to inflateEnd followed by inflateInit.
+            /// The stream will keep attributes that may have been set by inflateInit2.
+            /// </summary>
+            [SecurityCritical]
+            public ErrorCode InflateReset(int windowBits)
```

Comment:
This pattern of putting code in a finally block like this is done to deal with thread aborts, in particular because thread aborts are delayed across finally's so you don't have to worry about them interrupting a finally block.  However, thread aborts aren't a concern for .NET Core, and even if this library ends up being used on the desktop, it's very unlikely the library is entirely hardened against them.  I wouldn't worry about using try/finally's in this way.